src/modules/person/domain/username-generator.service.ts

```diff
@@ -1,44 +1,98 @@
-import { DomainError, EntityNotFoundError } from '../../../shared/error/index.js';
+import {
+    DomainError,
+    EntityNotFoundError,
+    InvalidNameError,
+    InvalidCharacterSetError,
+    InvalidAttributeLengthError,
+} from '../../../shared/error/index.js';
+
+// Regex to validate DIN-19379A for a string
+// Generated using the list of legal characters using these steps:
+// 1. All single-sequence characters are combined as best as possible.
+// 2. All two-sequence characters are grouped by their first codepoint
+// 3. All three-sequence characters are kept as is and appended at the end
+const DIN_19379A: RegExp =
+    /^(\u0020|\u0027|[\u002C-\u002E]|[\u0041-\u005A]|[\u0060-\u007A]|\u007E|\u00A8|\u00B4|\u00B7|[\u00C0-\u00D6]|[\u00D8-\u00F6]|[\u00F8-\u017E]|[\u0187-\u0188]|\u018F|\u0197|[\u01A0-\u01A1]|[\u01AF-\u01B0]|\u01B7|[\u01CD-\u01DC]|[\u01DE-\u01DF]|[\u01E2-\u01F0]|[\u01F4-\u01F5]|[\u01F8-\u01FF]|[\u0212-\u0213]|[\u0218-\u021B]|[\u021E-\u021F]|[\u0227-\u0233]|\u0259|\u0268|\u0292|[\u02B9-\u02BA]|[\u02BE-\u02BF]|\u02C8|\u02CC|[\u1E02-\u1E03]|[\u1E06-\u1E07]|[\u1E0A-\u1E11]|\u1E17|[\u1E1C-\u1E2B]|[\u1E2F-\u1E37]|[\u1E3A-\u1E3B]|[\u1E40-\u1E49]|[\u1E52-\u1E5B]|[\u1E5E-\u1E63]|[\u1E6A-\u1E6F]|[\u1E80-\u1E87]|[\u1E8C-\u1E97]|\u1E9E|[\u1EA0-\u1EF9]|\u2019|\u2021|\u0041\u030B|\u0043[\u0300\u0304\u0306\u0308\u0315\u0323\u0326]|\u0044\u0302|\u0046[\u0300\u0304]|\u0047\u0300|\u0048[\u0304\u0326\u0331]|\u004A[\u0301\u030C]|\u004B[\u0300\u0302\u0304\u0307\u0315\u031B\u0326]|\u004C[\u0302\u0325\u0326]|\u004D[\u0300\u0302\u0306\u0310]|\u004E[\u0302\u0304\u0306\u0326]|\u0050[\u0300\u0304\u0315\u0323]|\u0052[\u0306\u0325]|\u0053[\u0300\u0304\u0331]|\u0054[\u0300\u0304\u0308\u0315\u031B]|\u0055\u0307|\u005A[\u0300\u0304\u0306\u0308\u0327]|\u0061\u030B|\u0063[\u0300\u0304\u0306\u0308\u0315\u0323\u0326]|\u0064\u0302|\u0066[\u0300\u0304]|\u0067\u0300|\u0068[\u0304\u0326]|\u006A\u0301|\u006B[\u0300\u0302\u0304\u0307\u0315\u031B\u0326]|\u006C[\u0302\u0325\u0326]|\u006D[\u0300\u0302\u0306\u0310]|\u006E[\u0302\u0304\u0306\u0326]|\u0070[\u0300\u0304\u0315\u0323]|\u0072[\u0306\u0325]|\u0073[\u0300\u0304\u0331]|\u0074[\u0300\u0304\u0315\u031B]|\u0075\u0307|\u007A[\u0300\u0304\u0306\u0308\u0327]|\u00C7\u0306|\u00DB\u0304|\u00E7\u0306|\u00FB\u0304|\u00FF\u0301|\u010C[\u0315\u0323]|\u010D[\u0315\u0323]|\u0113\u030D|\u012A\u0301|\u012B\u0301|\u014D\u030D|\u017D[\u0326\u0327]|\u017E[\u0326\u0327]|\u1E32\u0304|\u1E33\u0304|\u1E62\u0304|\u1E63\u0304|\u1E6C\u0304|\u1E6D\u0304|\u1EA0\u0308|\u1EA1\u0308|\u1ECC\u0308|\u1ECD\u0308|\u1EE4[\u0304\u0308]|\u0043\u0328\u0306|\u004B\u035F\u0048|\u004B\u035F\u0068|\u004C\u0325\u0304|\u0052\u0325\u0304|\u0053\u031B\u0304|\u0063\u0328\u0306|\u006B\u035F\u0068|\u006C\u0325\u0304|\u0072\u0325\u0304|\u0073\u031B\u0304)*$/;
+
+// Specific replacements for german, danish and french
+const NORMALIZATION_LIST: { pattern: RegExp; base: string }[] = [
```

Comment:
Eventuell in eigene Datei Packen

Reply:
Die Ersetzungen gehören speziell zum username-generator